kaiax/reward/README.md

```diff
@@ -0,0 +1,172 @@
+    - The proposer's minting reward is fixed to a product of minting amount (M), validator's reward ratio (g) and KIP-82 proposer ratio (p). This amount is considered the minimum operation cost of a validator.
+    - Among the fees (F), half is always burnt since Magma. The other half (F/2) is burnt up to the proposer's minting reward (gpM), but the exceeding part (F/2 - gpM) is granted to the proposer.
+    - Summing up, the proposer is guaranteed a minimum even if transaction fees are no enough to support the validator operating cost, yet incentivized to include as many transactions as possible for more reward.
+
+![kip82_reward](./kip82_reward.png)
+
+## Persistent schema
+
+ This module does not persist any data.
+
+## In-memory structures
+
+### RewardConfig
+
+`RewardConfig` is a refined reward-related parameters.
+
+### RewardSummary
+
+```go
+type RewardSummary struct {
+	Minted   *big.Int `json:"minted"`
+	TotalFee *big.Int `json:"totalFee"`
+	BurntFee *big.Int `json:"burntFee"`
+}
+```
+
+`RewardSummary` is a summary of reward components. Calculation of it does not require StakingInfo so it can be calculated in full nodes. Useful for tracking the total supply of the native token.
+
+### RewardSpec
+
+```go
+type RewardSpec struct {
+	RewardSummary
+	Proposer *big.Int                    `json:"proposer"`
+	Stakers  *big.Int                    `json:"stakers"`
+	KIF      *big.Int                    `json:"kif"`
+	KEF      *big.Int                    `json:"kef"`
+	Rewards  map[common.Address]*big.Int `json:"rewards"`
+}
+```
+
+`RewardSpec` is a reward distribution specification for a block. Describes each reward component and their recipients.
+- It can represent the deferred reward to be distributed at the end of the block (e.g. FinalizeBlock).
+- It can also represent the block reward including both non-deferred and deferred fees (e.g. kaia_getReward API).
+
+### RewardResponse
+
+`RewardResponse` is a response type for `kaia_getReward` API.
+
+### AccumulatedRewardResponse
+
+`AccumulatedRewardResponse` is a response type for `kaia_getRewardsAccumulated` API.
+
+## Module lifecycle
+
+### Init
+
+- Dependencies:
+  - ChainDB: Raw key-value interface to access this module's persistent schema.
+  - ChainConfig: Holds the genesis configuration.
+  - Chian: Access to blocks, transactions and receipts.
+  - kaiax/gov: Access to the governance parameters.
+
+### Start and stop
+
+This module does not have any background threads.
+
+## Block processing
+
+### Consensus
+
+#### FinalizeHeader
+
+After all transactions are processed, calculate and accredit the block reward.
+
+### Execution
+
+#### PostRunTx
+
+When the non-deferred fees method is used, the transaction fees are accredited to the proposer after a transaction is executed.
+
+### Rewind
+
+This module does not have any persistent data to delete.
+
+## APIs
+
+### kaia_getReward
+
+### kaia_getRewardsAccumulated
```

Comment:
This is implemented in `gov_getRewardsAccumulated`; I remember it's to hide computation-intensive API.

Is `kaia_getTotalSupply` not being migrated?

Reply:
You're right. It must be `kaia_getReward` and `governance_getRewardsAccumulated`.

Yesterday I wanted to separate out totalSupply API as another module, but on second thought it is okay to be in rewards module. Let me add it here.

Reply:
Fixed README in a22f870.